### DISCUSSION TITLE
Artwork

### DISCUSSION THREAD

**Discussion by dleggo** [ravynsoft/ravynos#449] [General]
Some possible artwork. I can change anything if needed. I would've made a PR for this, but I'm not quite sure where they should go. Maybe in ```WindowServer```, or ```AppKit```?

![-](https://github.com/user-attachments/assets/b45ddc84-c0de-43ab-a6c0-bdb894270da9)
![png](https://github.com/user-attachments/assets/68681e79-ed39-4397-9bc7-3134c77f0238)
![png](https://github.com/user-attachments/assets/247c596f-c2e9-4513-98c8-41a784cf707f)
![Green](https://github.com/user-attachments/assets/5a5b2f92-649e-44f6-83ea-defd189918ba)
![mouse-arrow](https://github.com/user-attachments/assets/9fbcfa2e-acf1-4c83-aed7-62885830e7e2)
![Red](https://github.com/user-attachments/assets/cfe402e2-f933-499f-8f67-865bd886e1c0)
![X](https://github.com/user-attachments/assets/e228325f-408e-4f65-a98a-4dbab1235780)
![Yellow](https://github.com/user-attachments/assets/f131497c-4f5a-419d-a605-bad4a084598a)
An example of how those buttons look on a window.
![Sample](https://github.com/user-attachments/assets/905ac8dd-6061-4d45-bd3e-8a2730f1a939)

**Comment by mszoek** [comment]
Very nice, thank you!

**Comment by mszoek** [comment]
@dleggo can you please make sure all the pngs have the same spacing top, bottom, left, and right of the images? I think the yellow button is slightly lower than the others.

**Reply by dleggo** [reply]
Here is the image with a line at a fixed height.

![ImgWithLine](https://github.com/user-attachments/assets/03a23433-457e-49c0-9035-8c469da90d8a)

They all seem to be at the same height, unless if I am perceiving something incorrectly?
I also checked my images and they seemed to be lined up.

**Reply by mszoek** [reply]
Thanks. Might just be an illusion on my screen then. It seemed 1-2px lower at the top of the yellow circle than the red and green ones.

**Comment by mszoek** [comment]
For the record, here is what the icons look like in WindowServer :)

![normalwindow](https://github.com/user-attachments/assets/7f373042-f529-402c-ba0a-5e070cd183b0)
![controlhover](https://github.com/user-attachments/assets/9263ada6-4fe8-408d-bf5e-b7e0b67cf77a)

**Reply by dleggo** [reply]
Thanks! Is it possible to render the buttons with anti-aliasing? Does ```WindowServer``` support that yet? The maximize button seems to render with the arrows a bit small/deformed, which I never noticed during my tests, but maybe that is an anti-aliasing thing...? If needed, I can change anything to try make them render better too!

**Reply by mszoek** [reply]
It would be best to make them render properly at this size. I can implement AA eventually but I'm trying to keep any compute in the rendering loop very minimal since it is all software (no GPU accel) on the framebuffer backend.

**Reply by dleggo** [reply]
Unfortunately I can not reproduce this issue... When I scale the png to 16x16 (without anti-aliasing) I get.
![16x16](https://github.com/user-attachments/assets/5beced97-a80f-4f8e-8fb7-0e15198a7875)
How are the buttons being rendered? That might help me to find what I need to change. I could apply a blur effect to simulate anti-aliasing at small sizes... If needed I can also supply the svgs too.

**Reply by mszoek** [reply]
Hmm. The rendering just uncompresses the PNG and draws the bitmap. It is probably the scaling algorithm that causes this. I'll take a look. Can you try giving me 16x16 and 32x32 versions that looks ok to you? It'll be a good test.

**Reply by dleggo** [reply]
Both scaled in GIMP with no interpolation.

16x16
![16x16](https://github.com/user-attachments/assets/69fcc605-1f82-4f8f-bd2a-7c657a59223d)
32x32
![32x32](https://github.com/user-attachments/assets/5887abd2-aa16-42bb-92e1-8402cbf1b3d5)

Thanks for all your work on this!
